master/buildbot/worker/base.py

```diff
@@ -229,6 +229,19 @@ def _applyWorkerInfo(self, info):
         self.worker_status.setAccessURI(info.get("access_uri", None))
         self.worker_status.setVersion(info.get("version", "(unknown)"))
 
+        # import any os_<key>
```

Comment:
I think we should remove that part for this PR.

Ideally, we should create a renderable which retrive these from worker info dynamically, instead of adding a lot of useless properties in every build

Reply:
```
Interpolate("%(worker:os_id)s")
```

Reply:
wonder if it would make more sense to just store the info variable as self.info for renderable access?

Reply:
I am not sure how to implement this, is there some example somewhere how this could be done?

Reply:
we indeed will need to cache the info in self.worker.info.

Then Interpolate implementation is a bit tricky:

https://github.com/buildbot/buildbot/blob/b0875d0d017d25665511ad14f5d64887d4865cd8/master/buildbot/process/properties.py#L606

the parser is defined automatically given the keyword, so you need to implement a method named _parse_worker, to support Interpolate("%(worker:os_id)s")

This method should return an IRenderable which will lazily evaluate the info.

Reply:
Ok. Now it allows actually accessing any worker info field. I am not entirely sure if I am storing the info fields correctly into the Properties object, as in, do I need to do the bytes2unicode before calling it or could I just pass the values as-is into it?